From d235f2cdc156a232ebdde185e11381696d784356 Mon Sep 17 00:00:00 2001
From: Tim Orling <tim.orling@konsulko.com>
Date: Thu, 29 Feb 2024 14:49:49 -0800
Subject: [PATCH] conftest.py: double Hypothesis deadline

The deadline is by default 200ms, but this is intended to be useful to
the developer and not necessarily recommended for heavily loaded CI
systems. Avoid warnings by doubling the deadline. If we continue to
see problems, we can revisit disabling (deadline=None) or a longer
deadline.

https://github.com/HypothesisWorks/hypothesis/issues/3713
https://hypothesis.readthedocs.io/en/latest/settings.html#hypothesis.settings.deadline
https://lists.openembedded.org/g/openembedded-core/topic/104640034#196437

Fixes:
"""
Unreliable test timings! On an initial run, this test took 268.29ms,
which exceeded the deadline of 200.00ms, but on a subsequent run it
took 2.63 ms, which did not. If you expect this sort of variability in
your test timings, consider turning deadlines off for this test by
setting deadline=None.
"""

Upstream-Status: Submitted [https://github.com/python-attrs/attrs/pull/1252]

Signed-off-by: Tim Orling <tim.orling@konsulko.com>
---
 conftest.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/conftest.py b/conftest.py
index 144e5f3..242f02e 100644
--- a/conftest.py
+++ b/conftest.py
@@ -20,7 +20,9 @@ def _frozen(request):
 def pytest_configure(config):
     # HealthCheck.too_slow causes more trouble than good -- especially in CIs.
     settings.register_profile(
-        "patience", settings(suppress_health_check=[HealthCheck.too_slow])
+        "patience",
+        settings(suppress_health_check=[HealthCheck.too_slow],
+        deadline=timedelta(milliseconds=400))
     )
     settings.load_profile("patience")
 
-- 
2.34.1

